src/rules/selector-no-type/__tests__/index.js

```diff
@@ -217,3 +217,28 @@ testRule(rule, {
     description: "ignore less interpolation inside @for for nested descendant",
   }],
 })
+
+testRule(rule, {
+  ruleName,
+  config: [ true, { ignoreTypes: [ "fieldset", "/^my-/" ] } ],
+  skipBasicChecks: true,
+
+  accept: [ {
```

Comment:
Can we focus these on *just* the CSS that is being tested e.g.

```js
testRule(rule, {
  ruleName,
  config: [ true, { ignoreTypes: [ "fieldset", "/^my-/" ] } ],
  skipBasicChecks: true,

  accept: [ {
    code: "fieldset {}",
  }, {
    code: "my-type {}",
  }, {
    code: "my-other-type {}",
  } ],

  reject: [ {
    code: "a {}",
    line: 1,
    column: 1,
  }, {
    code: "not-my-type {}",
    message: messages.rejected,
    line: 1,
    column: 1,
  } ],
})
```

Here's we testing:

- 1st: checking string input
- 2nd & 3rd - checking regex input
- 4th - checking simple type selectors still flag
- 5th - checking regex is working